arroyo/processing/strategies/streaming/transform.py

```diff
@@ -225,12 +231,40 @@ def parallel_transform_worker_initializer(
         custom_initialize_func()
 
 
+@dataclass
+class ParallelTransformResult(Generic[TTransformed]):
+    """
+    Results of applying the transform function to a batch
+    of messages in parallel.
+
+    `index_processed_til` holds the index of the next
```

Comment:
nit: `index_processed_until` vs `index_processed_til`. Actually, `until` has an inclusive connotation to me (as in message was already processed) but it actually hasn't been.

Though I'm not really sure if it's really needed at all? This implementation would be simpler without it. What kind of scenario would actually cause this to happen?

Reply:
I'm not sure what to call it, it was just called `i` before and I wanted to make it more clear what this magic index meant. Maybe something like `next_index_to_process`?

Explanation in prev comment.

Reply:
Ah ok. It's called `start_index` in the parallel transform function